### PR TITLE
📚 Spell Check Documentation

### PR DESCRIPTION
**Pull request by YazeedAlKhalaf**
# 📚 Spell Check Documentation

This Pull Request has spell checking and correction of documentations of all packages!

### CODE REVIEW

**Comment by YazeedAlKhalaf** [comment]
@FilledStacks

**Comment by YazeedAlKhalaf** [comment]
@FilledStacks 
Fixed The Issue, Now It Is Possible to merge.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-- **Services**: A wrapper of a single functionality / feature set. This is commonly used to wrap things like showing a dialog, wrapping database functionality, integrating an api, etc.
+- **View**: Shows the UI to the user. Single widgets also qualify as views (for consistency in terminology) a view, in this case, is not a "Page" it's just a UI representation.
+- **ViewModel**: Manages the state of the View, business logic, and any other logic as required from user interaction. It does this by making use of the services
+- **Services**: A wrapper of a single functionality/feature set. This is commonly used to wrap things like showing a dialog, wrapping database functionality, integrating an API, etc.
 
-Lets go over some of those principles to follow during development.
+Let's go over some of those principles to follow during development.
 
-- Views should never MAKE USE of a service directly.
+- Views should never MAKE USE of service directly.
```
Comment (FilledStacks) [inline README.md]: add back the 'a'

File: README.md [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 - Views should contain zero to (preferred) no logic. If the logic is from UI only items then we do the least amount of required logic and pass the rest to the ViewModel.
 - Views should ONLY render the state in its ViewModel.
-- ViewModels for widgets that represent page views are bound to a single View only.
-- ViewModels may be re-used if the UI required the exact same functionality.
+- ViewModels for widgets that represent page views are bound to a single view only.
```
Comment (FilledStacks) [inline README.md]: Add back the capital for View. It's a name in this case, the name of one of the three components in the state management architecture so it starts with a capital

File: README.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+- ViewModels may be re-used if the UI required the same functionality.
```
Comment (FilledStacks) [inline README.md]: Change required to require

File: packages/stacked/README.md [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-Lets go over some of those principles to follow during development.
+Let's go over some of those principles to follow during development.
 
-- Views should never MAKE USE of a service directly.
+- Views should never MAKE USE of services directly.
 - Views should contain zero to (preferred) no logic. If the logic is from UI only items then we do the least amount of required logic and pass the rest to the ViewModel.
 - Views should ONLY render the state in its ViewModel.
-- ViewModels for widgets that represent page views are bound to a single View only.
-- ViewModels may be re-used if the UI required the exact same functionality.
+- ViewModels for widgets that represent page views are bound to a single view only.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: Change back to capital View

File: packages/stacked/README.md [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+- ViewModels may be re-used if the UI required the same functionality.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: change required to require

File: packages/stacked/README.md [hunk 6]
```diff
@@ -331,7 +331,7 @@ class DuplicateNameWidget extends ViewModelWidget<Human> {
 
 ### Non reactive ViewModelWidget
 
-Sometimes you want a widget to have access to the ViewModel but you don't want it to rebuild when notifyListeners is called. In this case you can set the reactive value to false for the super constructor of the `ViewModelWidget`. This is commonly used in widgets that don't make use of the models state and only it's functionality.
+Sometimes you want a widget to have access to the ViewModel but you don't want it to rebuild when `notifylisteners()` is called. In this case, you can set the reactive value to false for the super constructor of the `ViewModelWidget`. This is commonly used in widgets that don't make use of the model's state and only it's functionality.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: notifyListeners should have a capital L

File: packages/stacked/README.md [hunk 7]
```diff
@@ -423,19 +423,19 @@ All the major functionality for the BaseViewModel is shown above
 
 ## Reactivity
 
-One thing that was common a scenario with the first implementation of this architecture that was clearly lacking is reacting to values changed by different ViewModels. I don't have the exact implementation that I would hope for but without reflection some things will have to be a bit more verbose. The stacked architecture makes provision for ViewModels to react to changes to values in a service by making use of RxValue from the [Observable-Ish](https://pub.dev/packages/observable_ish) package.
+One thing that was common in a scenario with the first implementation of this architecture that was lacking is reacting to values changed by different ViewModels. I don't have the exact implementation that I would hope for but without reflection, some things will have to be a bit more verbose. The stacked architecture makes provision for ViewModels to react to changes to values in service by making use of RxValue from the [Observable-Ish](https://pub.dev/packages/observable_ish) package.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: please remove "that was lacking" as well

File: packages/stacked/README.md [hunk 8]
```diff
@@ -458,14 +458,14 @@ class InformationService with ReactiveServiceMixin { //1
 }
 ```
 
-Easy peasy. This service can now be listened too when any of the properties passed into the `listenToReactiveValues` is changed. So how do listen to these values? I'm glad you asked. Lets move onto the `ReactiveViewModel`.
+Easy peasy. This service can now listen too when any of the properties passed into the `listenToReactiveValues` is changed. So how do you listen to these values? I'm glad you asked. Let's move onto the `ReactiveViewModel`.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: add back "be listened" to. The service is not listening to itself. The ViewModel is listening to it.

File: packages/stacked/README.md [hunk 9]
```diff
@@ -458,14 +458,14 @@ class InformationService with ReactiveServiceMixin { //1
...
 
 ### Reactive View Model
 
-This ViewModel extends the `BaseViewModel` and adds an additional function that allows you to listen to services that are being used in the model. There are two thing you have to do to make a ViewModel react to changes in a service.
+This ViewModel extends the `BaseViewModel` and adds a function that allows you to listen to services that are being used in the model. There are two things you have to do to make a ViewModel react to changes in a service.
 
 1. Extend from `ReactiveViewModel`.
-2. Implement reactiveServices getter that return a list of reactive services.
+2. Implement reactive services getter that returns a list of reactive services.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: please make this one word again using camel case, it refers to the code to be used. Change returns back to return

File: packages/stacked/README.md [hunk 10]
```diff
@@ -539,7 +539,7 @@ Besides having the onError function you can override the `ViewModel` will also s
 
 ### FutureViewModel
 
-This `ViewModel` extends the `BaseViewModel` to provide functionality to easily listen to a Future that fetches data. This requirement came off a Details view that has to fetch additional data to show to the user after selecting an item. When you extend the `FutureViewModel` you can provide a type which will then require you to override the future getter where you can set the future you want to run.
+This `ViewModel` extends the `BaseViewModel` to provide the functionality to easily listen to a Future that fetches data. This requirement came off a Details view that has to fetch additional data to show to the user after selecting an item. When you extend the `FutureViewModel` you can provide a type which will then require you to override the future getter where you can set the future you want to run.
```
Comment (FilledStacks) [inline packages/stacked/README.md]: remove the here

File: packages/stacked_hooks/README.md [hunk 11]
```diff
@@ -4,7 +4,7 @@ This package contains widgets that allow you to use the Flutter Hooks package wi
 
 ## HookViewModelWidget
 
-The `ViewModelWidget` is an implementation of a widget class that returns a value provided by Provider as a parameter in the build function of the widget. This allows for easier consumption and use of viewmodels without boilerplate. The `HookViewModelWidget` allows you to use this widget and make use of Flutter Hooks inside the build function. This is very useful when you want to use `TextEditting` controllers and you're implementing this architecture.
+The `ViewModelWidget` is an implementation of a widget class that returns a value provided by the Provider as a parameter in the build function of the widget. This allows for easier consumption and use of ViewModel without boilerplate. The `HookViewModelWidget` allows you to use this widget and make use of Flutter Hooks inside the build function. This is very useful when you want to use `TextEditting` controllers and you're implementing this architecture.
```
Comment (FilledStacks) [inline packages/stacked_hooks/README.md]: Remove "the" here. Provider refers to the package not the object.

File: packages/stacked_services/README.md [hunk 12]
```diff
@@ -71,7 +71,7 @@ await _dialogService.showDialog(
 
 ### Custom Dialog UI
 
-In addition to platform specific UI you can also build your own custom dialog. To do that we'll do the following. In your UI folder or shared folder under UI, if you have one, create a new file called `setup_dialog_ui.dart`. Inside you will create a new function called `setupDialogUi`. In there you will call the function `registerCustomDialogUi` on the `DialogService`. _Look at the `setup_dialog_ui` file for a full example_
+In addition to platform-specific UI, you can also build your custom dialog. To do that we'll do the following. In your UI folder or shared folder under UI, if you have one, create a new file called `setup_dialog_ui.dart`. Inside you will create a new function called `setupDialogUi`. In there you will call the function `registerCustomDialogUi` on the `DialogService`. _Look at the `setup_dialog_ui` file for a full example_
```
Comment (FilledStacks) [inline packages/stacked_services/README.md]: replace "your" with a if you're removing "own"

File: packages/stacked_services/README.md [hunk 13]
```diff
@@ -148,7 +148,7 @@ Now in your ViewModels you can make use of the dialog as follows.
 
 ### Returning Data from Custom Dialog
 
-The custom dialog follows the same rules as the normal dialog. Calling `completeDialog` and passing in a `DialogResponse` object will return it to the caller that's awaiting on the dialog response UI. So when you have a tap handler in your dialog and you want to close the dialog, use the `completeDialog` function.
+The custom dialog follows the same rules as the normal dialog. Calling `completeDialog` and passing in a `DialogResponse` object will return it to the caller that's waiting on the dialog response UI. So when you have a tap handler in your dialog and you want to close the dialog, use the `completeDialog` function.
```
Comment (FilledStacks) [inline packages/stacked_services/README.md]: change back to awaiting. It referes to the async/await functionality. not waiting in the normal life sense

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Please make the changes to your updates.

File: packages/stacked/README.md (hunk 9 above)
Comment (YazeedAlKhalaf) [reply packages/stacked/README.md]: But `returns` is correct here grammatically

File: packages/stacked_services/README.md (hunk 13 above)
Comment (YazeedAlKhalaf) [reply packages/stacked_services/README.md]: Ah okay 😆

**Comment by YazeedAlKhalaf** [comment]
All Changes are Done! 😊🚀

File: README.md [hunk 14]
```diff
@@ -1,24 +1,24 @@
...
-- **Services**: A wrapper of a single functionality / feature set. This is commonly used to wrap things like showing a dialog, wrapping database functionality, integrating an api, etc.
+- **View**: Shows the UI to the user. Single widgets also qualify as views (for consistency in terminology) a view, in this case, is not a "Page" it's just a UI representation.
+- **ViewModel**: Manages the state of the View, business logic, and any other logic as required from user interaction. It does this by making use of the services
+- **Services**: A wrapper of a single functionality/feature set. This is commonly used to wrap things like showing a dialog, wrapping database functionality, integrating an API, etc.
 
-Lets go over some of those principles to follow during development.
+Let's go over some of those principles to follow during development.
 
-- Views should never MAKE USE of a service directly.
+- Views should never MAKE USE a of service directly.
```
Comment (FilledStacks) [inline README.md]: This sentence doesn't make sense. Change back to 

Views should never MAKE USE of a service directly. 'a of' doesn't bear meaning in english

**Review by FilledStacks — CHANGES_REQUESTED** [review]
change back the  of a that was swapped around to a of.

File: README.md (hunk 14 above)
Comment (YazeedAlKhalaf) [reply README.md]: yea I see 😂🤣

**Review by RodolfoSilva — APPROVED** [review]

**Comment by YazeedAlKhalaf** [comment]
@FilledStacks

**Comment by FilledStacks** [comment]
Thank you @YazeedAlKhalaf I appreciate the contributions! I'll push the read me updates to the packages with the next package release.

**Comment by YazeedAlKhalaf** [comment]
You're welcome

**Pull request merged**